pkg/nuget/parse.go

```diff
@@ -0,0 +1,84 @@
+package nuget
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"io"
+)
+
+// The lockfile that nuget uses (packages.lock.json) works basically like this:
+// Each 'build type' (for example .NETCoreApp,Version=v3.1) have a set of dependencies.
+// Those dependencies can be 'Direct', 'Transitive' and 'Project'
+// (meaning of those are quite self explanatory I think). The dependency consists of a string
+// value key (package name) and an object, from which we need to fetch a few different values.
+// The following values should be parsed right away: resolved (resolved version),
+// type (this information might be worth having) and finally, it's dependencies.
+// The dependencies in this stage of the file are only a string map where the key is package name
+// and value is package version.
+
+type LockFile struct {
+	Version      int
+	Dependencies map[string]map[string]Dependency
+}
+
+type Dependency struct {
+	Type         string            `json:"type"`
+	Resolved     string            `json:"resolved"`
+	Dependencies map[string]string `json:"dependencies"`
+	ContentHash  string            `json:"contentHash"`
+}
+
+func Parse(r io.Reader) ([]types.Library, error) {
+	var lockFile LockFile
+	decoder := json.NewDecoder(r)
+	err := decoder.Decode(&lockFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var libraries []types.Library
+	unique := map[string]struct{}{}
+
+	// The first loop is the targets (.NETCoreApp,Version=v3.1)
+	// that means that it's not really a package, but there are dependencies
+	// under said target.
+	for _, targetContent := range lockFile.Dependencies {
+		// Here comes the 'real' dependencies.
+		for topPkgName, topPkgContent := range targetContent {
+			// And inside the dependency, there are another level of dependencies.
+			// But that's actually it, I swear!
```

Comment:
It's pretty self documenting as code so you can take off the comments. Appreciate your commentary though!